### PR TITLE
Improve use of self

### PR DESCRIPTION
**Pull request by 3lvis**
"Fixes" #115

>Should we continue this discussion? We still have different thoughts about that and also we've got some issues with using of `self` in the debugger. I guess @zenangst could describe it in details.  
— @markvaldy

### CODE REVIEW

File: style-guidelines/Swift.md [hunk 1]
```diff
@@ -146,15 +146,15 @@ Unless you require functionality that can only be provided by a class (like iden
...
 
-Use `self` when referencing global variables like properties:
+Use `self` only when *needed*:
 
 ```swift
 class BoardLocation {
   let row: Int, column: Int, foo: String
 
   init(row: Int, column: Int) {
-    self.row = row
```
Comment (kostiakoval) [inline style-guidelines/Swift.md]: you need self here actually :) so roll this change back

Comment (3lvis) [inline style-guidelines/Swift.md]: Done :)

**Comment by 3lvis** [comment]
@hyperoslo/ios any other place?

**Comment by 3lvis** [comment]
So, I think we should do this. At least for now and evaluate if this is something that we *actually* need.

**Comment by 3lvis** [comment]
Last time we where 3.5 (self.yes) and 1.5 (just yes, at least until the compiler begs you to add self).

So, @RamonGilabert, @zenangst, @jgorset * 0.5 and me where in favor of keeping `self`.

I'm voting against so that makes it.

2.5 vs 2.5!

**Comment by 3lvis** [comment]
Anyway, if this is good feel free to merge :)

**Comment by zenangst** [comment]
My swing vote as "swong", I think we should remove it.

**Comment by fespinoza** [comment]
I like the fact that I can learn swift on my own and I already know how to write my code given your conventions :D

**Comment by jgorset** [comment]
Mine has also swong — I'm an entire 1.0 in favour of dropping `self`now. :-)

**Comment by zenangst** [comment]
:hammer: BANG!

**Comment by vadymmarkov** [comment]
:tada:

**Comment by kostiakoval** [comment]
:tada: ))

**Pull request merged**